src/FormidableError.js

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable no-plusplus */
+
+let errorCodes = 0;
+const missingPlugin = errorCodes++;
+const pluginFunction = errorCodes++;
+const aborted = errorCodes++;
+const noParser = errorCodes++;
+const uninitializedParser = errorCodes++;
+const filenameNotString = errorCodes++;
+const maxFieldsSizeExceeded = errorCodes++;
+const maxFieldsExceeded = errorCodes++;
+const smallerThanMinFileSize = errorCodes++;
+const biggerThanMaxFileSize = errorCodes++;
+const noEmptyFiles = errorCodes++;
+const missingContentType = errorCodes++;
+const malformedMultipart = errorCodes++;
+const missingMultipartBoundary = errorCodes++;
+const unknownTransferEncoding = errorCodes++;
```

Comment:
This is an interesting pattern.

Why not just using string based codes?

Something like:

```javascript
const missingPlugin = "ERR_MISSING_PLUGIN";
const pluginFunction = "ERR_PLUGIN_FUNCTION";
// ...
```

I think this might help with debugging errors as they happen and it might make it easier to add new errors or deprecate existing ones in the future.

Reply:
It might be worth considering using `Symbol` if you want to force users to use the internal representations for comparison. This is something that can help to avoid breaking changes in case you decide to change the internal representation of these constants for whatever reason.

An example:

```javascript
const MISSING_PLUGIN = Symbol('ERR_MISSING_PLUGIN');
const PLUGIN_FUNCTION = Symbol('ERR_PLUGIN_FUNCTION');
```

Than user code would have to do:

```javascript
const { FormidableError } = require('formidable')

// ...

form.parse(req, (err) => {
  if (err.code === FormidableError. missingPlugin) {
    // ...
  }
})
```

Reply:
Symbols are nice but if you want to log the errors in a file, it can't be stringified. 
Numbers are better for performance than string for comparison. There is already a string in the message if you need a string.